.github/workflows/build-deploy.yml

```diff
@@ -8,6 +8,10 @@ permissions:
   id-token: write
   contents: read # This is required for actions/checkout
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ !contains(github.ref, 'staging')}}
```

Comment:
Approved, but is there any particular reason why in-progress `staging` pipeline shouldn't get cancelled when a new one starts?

Reply:
Thanks for catching!